akka-actor-typed/src/main/scala/akka/actor/typed/ActorRefResolver.scala

```diff
@@ -76,7 +76,10 @@ abstract class ActorRefResolver extends Extension {
   }
 
   override def resolveActorRef[T](serializedActorRef: String): ActorRef[T] =
-    classicSystem.provider.resolveActorRef(serializedActorRef)
+    if (IgnoreActorRef.isIgnoreRefPath(serializedActorRef))
+      classicSystem.toTyped.ignoreRef
+    else
+      classicSystem.provider.resolveActorRef(serializedActorRef)
 }
```

Comment:
I understand and agree, but the problem here is that if we move the impl to `resolveActorRef` only classic ref will be cached. 

`classicSystem.toTyped.ignoreRef` always returns the same instance that is already wrapped on an adapter. Otherwise, we will have a new adapter for each invocation.

Reply:
btw, `classic.provider.resolveActorRef` is also taking care of this. We have code duplication here to avoid the adapter instance.

Reply:
maybe slightly better performance (for ordinary refs) with this:
```scala
override def resolveActorRef[T](serializedActorRef: String): ActorRef[T] = {
  val ref = classicSystem.provider.resolveActorRef(serializedActorRef)
  if (ref eq classicSystem.ignoreRef)
    classicSystem.toTyped.ignoreRef // use cached ActorRefAdapter
  else
    ref
}
```